### PR TITLE
replaced telegram with an async version

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- use async communication when talking to telegram to avoid blocking sanic workers

### CODE REVIEW

File: rasa/core/channels/telegram.py
Comment (ancalita) [inline rasa/core/channels/telegram.py:43]: Docstring required.

**Review by ancalita — APPROVED** [review]
⚡ Nice one!

File: rasa/core/channels/telegram.py
Comment (ancalita) [reply rasa/core/channels/telegram.py:220]: Good point! It was this [test failing,](https://github.com/RasaHQ/rasa/blob/a032eed9cc9ec36c1f2b0848122c88987884b195/tests/core/channels/test_telegram.py#L25) so yes, i'll change to a json parser.

Comment (ancalita) [reply rasa/core/channels/telegram.py]: Resolved here: https://github.com/RasaHQ/rasa/pull/11552/commits/b8f96d60a6d662b67fbba262ab9f2bdd9f841880

**Review by tmbo — COMMENTED** [review]
this looks great, thanks a lot :+1: (I can't approve it since I created the PR but please go ahead and approve it)

**Comment by benos** [comment]
Hi, 

This is failing for me. I get the following error when rasa starts:

image

And the bot is unresponsive when I type something in Telegram. Downgrading to 3.2.9 fixes the problem.

My credentials look like this:

image

Thanks!

Ben

**Comment by ancalita** [comment]
@benos thanks for flagging! I'll open a fix PR asap for this on `3.3.x` and hopefully that should fix this issue.

**Comment by benos** [comment]
Hi @ancalita 

That sounds great, but do you have any idea when this fix might be made live? I don't see any change on 3.3.1

Thanks!

Ben

**Comment by ancalita** [comment]
@benos Apologies for the delay, other priorities took hold 🙈
I've opened [this PR](https://github.com/RasaHQ/rasa/pull/11818), once it gets reviewed and merged, I'll release a new patch 🤞🏻

**Comment by benos** [comment]
@ancalita Awesome! Many thanks

**Comment by benos** [comment]
Hi @ancalita,

Sorry to reopen this, but this change has created an additional bug. When triggering an intent like so:

http://localhost:5005/conversations/[redacted]/trigger_intent?output_channel=telegram

This fails with the following error:

{"version":"3.4.1","status":"failure","message":"An unexpected error occurred. Error: asyncio.run() cannot be called from a running event loop","reason":"ConversationError","details":{},"help":null,"code":500}

I assume get_output_channel is re-run by the triggering code.

Best,

Ben

**Comment by benos** [comment]
@ancalita While I'm at it, might this line cause a leak if called repeatedly? 

channel = TelegramOutput(self.access_token)

I've actually noticed a steadily rising CPU consumption which seems linked to calling trigger_intent, but could never really pinpoint the problem.

**Comment by ancalita** [comment]
@benos I'm so sorry to hear about these new difficulties 🙈 I didn't envisage the potential issues of this change when using the HTTP API. Could you please create a ticket in this [Jira board](https://rasa-open-source.atlassian.net/jira/software/c/projects/OSS/boards/1?quickFilter=5), the new home for OSS issues, with steps for reproducing the error.

> I've actually noticed a steadily rising CPU consumption which seems linked to calling trigger_intent

A separate ticket for this would be welcome too 🙏🏻 with the reference to the particular line in the codebase if possible. Then you can link these two tickets as related, so they can inform each other.

**Comment by benos** [comment]
Hi @ancalita ,

I created two tickets:

https://rasa-open-source.atlassian.net/jira/software/c/projects/OSS/boards/1?modal=detail&selectedIssue=OSS-677&quickFilter=5

https://rasa-open-source.atlassian.net/jira/software/c/projects/OSS/boards/1?modal=detail&selectedIssue=OSS-676&quickFilter=5

The silver lining could be that this might have revealed the origin of this CPU link that has been plaguing us since we launched.

Best,

Ben

**Pull request merged**
